src/modes/ModeEstimation.jl

```diff
@@ -18,7 +18,8 @@ export  constrained_space,
         MAP,
         MLE,
         OptimLogDensity,
-        OptimizationContext,
+        LikelihoodOptimizationContext,
```

Comment:
Could we avoid defining more contexts? 🤔 Or at least not expose it to users?

Reply:
Happy to not export; I just did it because it was already being done.

But it's worth pointing out that it's not being exported to the user; this is from the `ModeEstimation` module. You still want me to remove the export?

Reply:
Ideally, I would like to stick with a single OptimizationContext, that's my main complaint.

I think it should be possible to just combine it with the existing contexts. It seems orthogonal to LikelihoodContext etc., so it would be nice to combine them easily. This would also scale better than having dedicated optimization contexts for every other context you want to combine it with.

Reply:
> This would also scale better than having dedicated optimization contexts for every other context you want to combine it with.

In what sense do you mean here? Even if we allow it to wrap other contexts (as is done in the current release), the scaling is the same, no?

Reply:
I'd also just mention that this PR is somewhat urgent since it is silently breaking all MLE estimation with models using `condition` :confused:  So if we're talking about making improvements to how we deal with hierarchies of contexts _in general_ (i.e. beyond addressing this bug), we should defer that.

Reply:
> As in, what do you have in mind for

Sorry, I didn't have time to think about it more carefully earlier today. I opened a PR with what I imagined: https://github.com/TuringLang/Turing.jl/pull/2022

> In what sense do you mean here?

I thought about allowing all kind/types of other leaf contexts. But I guess you're right that it's not possible anyway (currently) (and also not supported in #2022).